ctapipe/tools/make_irf.py

```diff
@@ -0,0 +1,353 @@
+        ]
+
+        self.log.debug("True Energy bins: %s" % str(self.true_energy_bins.value))
+        self.log.debug("FoV offset bins: %s" % str(self.fov_offset_bins.value))
+        for label, mask in masks.items():
+            effective_area = effective_area_per_energy_and_fov(
+                self.signal_events[mask],
+                self.sim_info,
+                true_energy_bins=self.true_energy_bins,
+                fov_offset_bins=self.fov_offset_bins,
+            )
+            hdus.append(
+                create_aeff2d_hdu(
+                    effective_area[..., np.newaxis],  # +1 dimension for FOV offset
+                    self.true_energy_bins,
+                    self.fov_offset_bins,
+                    extname="EFFECTIVE AREA" + label,
+                )
+            )
+            edisp = energy_dispersion(
+                self.signal_events[mask],
+                true_energy_bins=self.true_energy_bins,
+                fov_offset_bins=self.fov_offset_bins,
+                migration_bins=self.energy_migration_bins,
+            )
+            hdus.append(
+                create_energy_dispersion_hdu(
+                    edisp,
+                    true_energy_bins=self.true_energy_bins,
+                    migration_bins=self.energy_migration_bins,
+                    fov_offset_bins=self.fov_offset_bins,
+                    extname="ENERGY_DISPERSION" + label,
+                )
+            )
+        # Here we use reconstructed energy instead of true energy for the sake of
+        # current pipelines comparisons
+        bias_resolution = energy_bias_resolution(
+            self.signal_events[self.signal_events["selected"]],
+            self.true_energy_bins,
+            bias_function=np.mean,
+            energy_type="true",
+        )
+        hdus.append(fits.BinTableHDU(bias_resolution, name="ENERGY_BIAS_RESOLUTION"))
+
+        # Here we use reconstructed energy instead of true energy for the sake of
+        # current pipelines comparisons
+        ang_res = angular_resolution(
+            self.signal_events[self.signal_events["selected_gh"]],
+            self.reco_energy_bins,
+            energy_type="reco",
+        )
+        hdus.append(fits.BinTableHDU(ang_res, name="ANGULAR_RESOLUTION"))
+
+        sel = self.background_events["selected_gh"]
+        self.log.debug("%d background events selected" % sel.sum())
+        self.log.debug("%f obs time" % self.obs_time)
+        background_rate = background_2d(
+            self.background_events[sel],
+            self.reco_energy_bins,
+            fov_offset_bins=self.fov_offset_bins,
+            t_obs=self.obs_time * u.Unit(self.obs_time_unit),
+        )
+        hdus.append(
+            create_background_2d_hdu(
+                background_rate,
+                self.reco_energy_bins,
+                fov_offset_bins=self.fov_offset_bins,
+            )
+        )
+
+        psf = psf_table(
```

Comment:
Compute PSF only if not pointlike.

Should be a configurable class so we can later extend to computing different variants.